crates/flipperzero/src/subghz/device.rs

```diff
@@ -0,0 +1,193 @@
+//! Safe wrapper utilizing the API provided in `lib/subghz/devices/devices.h`
+
+use super::{error::SubGhzError, preset::SubGhzPreset};
+use core::{ffi::CStr, ptr::null_mut};
+use flipperzero_sys::{
+    subghz_devices_begin, subghz_devices_deinit, subghz_devices_end, subghz_devices_flush_rx,
+    subghz_devices_flush_tx, subghz_devices_get_by_name, subghz_devices_idle, subghz_devices_init,
+    subghz_devices_is_connect, subghz_devices_is_frequency_valid,
+    subghz_devices_is_rx_data_crc_valid, subghz_devices_load_preset, subghz_devices_read_packet,
+    subghz_devices_reset, subghz_devices_rx_pipe_not_empty, subghz_devices_set_frequency,
+    subghz_devices_set_rx, subghz_devices_set_tx, subghz_devices_sleep,
+    subghz_devices_write_packet, SubGhzDevice,
+};
+use uom::si::{frequency::hertz, u32::Frequency};
+
+pub struct SubGhz {
+    device: &'static SubGhzDevice,
+}
+
+impl SubGhz {
+    /// For the internal device use "cc1101_int" (From `lib/subghz/devices/cc1101_int/cc1101_int_interconnect.c`)
+    pub fn subghz_devices_get_by_name(name: &CStr) -> Option<SubGhz> {
+        unsafe { subghz_devices_init() }
+
+        // Safety: input type enforeced by CStr type.
```

Comment:
I am not sure that this is sound: `CStr` only guarantees pointer validity and the absence of `\0`s, but it does *not* guarantee that the string is valid UTF-8 and (but I may be mistaken here) some Furi APIs require more specific strings.

Reply:
Based on what I saw, the string is eventually used by the following function: https://github.com/flipperdevices/flipperzero-firmware/blob/release/lib/subghz/registry.c#L3

My interpretation is that I just needed to make sure I had a null terminated array as the input. That being said, I was looking at commit 10 behind that I linked to, so I should probably look again since it looks like the file changed a fair bit.

Reply:
To add additional info, the string is compared with strcmp() to find the matching device in the list.

And no, CStr does not guarantee valid UTF-8 (it can be created from any &[u8] and the "constructors" only are checking for null termination.)

Reply:
Nice, given that it is only used for `strcmp`, `CStr` fits the purpose